salt/modules/cassandra_cql.py

```diff
@@ -0,0 +1,585 @@
+    :type  cql_user:       str
+    :param cql_pass:       The Cassandra user password if authentication is turned on.
+    :type  cql_pass:       str
+    :param port:           The Cassandra cluster port, defaults to None.
+    :type  port:           int
+    :return:               The session and cluster objects.
+    :rtype:                cluster object, session object
+    '''
+
+    contact_points = _loadproperties(property_name=contact_points, config_option='cluster')
+    contact_points = contact_points if isinstance(contact_points, list) else contact_points.split(',')
+    port = _loadproperties(property_name=port, config_option='port', set_default=True, default=9042)
+    cql_user = _loadproperties(property_name=cql_user, config_option='username', set_default=True, default="cassandra")
+    cql_pass = _loadproperties(property_name=cql_pass, config_option='password', set_default=True, default="cassandra")
+
+    try:
+        auth_provider = PlainTextAuthProvider(username=cql_user, password=cql_pass)
+        cluster = Cluster(contact_points, port=port, auth_provider=auth_provider)
+        session = cluster.connect()
+        log.debug('Successfully connected to Cassandra cluster at %s', contact_points)
+        return cluster, session
+    except (ConnectionException, ConnectionShutdown, NoHostAvailable):
+        log.error('Could not connect to Cassandra cluster at %s', contact_points)
+        raise CommandExecutionError('ERROR: Could not connect to Cassandra cluster.')
+
+
+def _query(query, contact_points=None, port=None, cql_user=None, cql_pass=None):
+    '''
+    Run a query on a Cassandra cluster and return a dictionary.
+
+    :param query:          The query to execute.
+    :type  query:          str
+    :param contact_points: The Cassandra cluster addresses, can either be a string or a list of IPs.
+    :type  contact_points: str | list[str]
+    :param cql_user:       The Cassandra user if authentication is turned on.
+    :type  cql_user:       str
+    :param cql_pass:       The Cassandra user password if authentication is turned on.
+    :type  cql_pass:       str
+    :param port:           The Cassandra cluster port, defaults to None.
+    :type  port:           int
+    :param params:         The parameters for the query, optional.
+    :type  params:         str
+    :return:               A dictionary from the return values of the query
+    :rtype:                list[dict]
+    '''
+    cluster, session = _connect(contact_points=contact_points, port=port, cql_user=cql_user, cql_pass=cql_pass)
+
+    session.row_factory = dict_factory
+    ret = []
+
+    try:
+        results = session.execute(query)
+    except BaseException as e:
+        log.error('Something went wrong when executing the following query: %s', query)
+        msg = "ERROR: Cassandra query failed: %s" % query
+        raise CommandExecutionError(msg, e)
+
+    if results:
```

Comment:
Since I am not sure what `session.execute(query)` supposed to return, then in case it is `None`, I would drop this `if` clause by doing the following:
```python
try
    results = session.execute(query) or []
except:
    ...
for result in results:
    ...
```